ttnn/cpp/ttnn/operations/pool/maxpool/maxpool_pybind.hpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "ttnn/cpp/pybind11/decorators.hpp"
+#include "ttnn/operations/pool/maxpool/max_pool.hpp"
+
+#include "ttnn/types.hpp"
+
+namespace py = pybind11;
+using array2_t = std::array<uint32_t, 2>;
+namespace ttnn {
+namespace operations {
+namespace maxpool {
+
+
+void py_module(py::module& module) {
+    module.def(
+        "max_pool2d",
+        &max_pool2d,
+        py::arg("input").noconvert(),
+        py::arg("in_n").noconvert(),
+        py::arg("in_h").noconvert(),
+        py::arg("in_w").noconvert(),
+        py::arg("kernel_h").noconvert(),
+        py::arg("kernel_w").noconvert(),
+        py::arg("stride_h") = 1,
+        py::arg("stride_w") = 1,
+        py::arg("pad_h") = 0,
+        py::arg("pad_w") = 0,
+        py::arg("dilation_h") = 1,
+        py::arg("dilation_w") = 1,
+        py::kw_only(),
+        py::arg("memory_config") = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
+        py::arg("nblocks") = 1,
+        py::arg("use_multicore") = true,
```

Comment:
Why do we need this flag?

Reply:
There are single core & multi-core versions of max pool. This flag switches between the two.

Reply:
single tensix-core version? where is it used? why do we need both?

Reply:
AFAIK, the single-core version supports certain scenarios, like having the input/output as interleaved, which the multi-core version doesn't.